### ISSUE TITLE
TSTypeAssertion's expression property can't be a TSTypeAssertion node

### ISSUE THREAD

**Issue by G-Rath** [typescript-eslint/typescript-eslint#802]
```ts
<string><unknown>null;
```

In [`AST Explorer`](https://astexplorer.net/#/gist/e6207d1eb70f6131eee1ee02fa04de44/aa47d0e2d7dcafcd298dba850f9f7e294131ca9b) the above code has the following structure:

```
Program.body
\ ExpressionStatement.expression
 \ TSTypeAssertion.expression
  \ TSTypeAssertion.expression
   \ Literal.value
    \ null
```

However, the types don't allow this:

```
// ts-estree.d.ts
export interface TSTypeAssertion extends BaseNode {
    type: AST_NODE_TYPES.TSTypeAssertion;
    typeAnnotation: TypeNode;
    expression: UnaryExpression;
}

export interface UnaryExpression extends UnaryExpressionBase {
    type: AST_NODE_TYPES.UnaryExpression;
}
```

This means I can't do:

```
// this is fine
interface AsExpressionChain<Expression extends TSESTree.Expression>
  extends TSESTree.TSAsExpression {
  expression: AsExpressionChain<Expression> | Expression;
}

// error!
interface TypeAssertionChain<Expression extends TSESTree.UnaryExpression>
  extends TSESTree.TSTypeAssertion {
  expression: TypeAssertionChain<Expression> | Expression;
}
```

error details:
```
Error:(23, 11) TS2430: Interface 'TypeAssertionChain<Expression>' incorrectly extends interface 'TSTypeAssertion'.
  Types of property 'expression' are incompatible.
    Type 'Expression | TypeAssertionChain<Expression>' is not assignable to type 'UnaryExpression'.
      Type 'TypeAssertionChain<Expression>' is missing the following properties from type 'UnaryExpression': operator, prefix, argument
```

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.13.0` |
| `TypeScript`                           | `3.5.3` |
| `node`                                 | `10.15.3` |
| `npm`                                  | `6.10.2` |

(the version of `@typescript-eslint/parser` on `ast` is `1.11.0`).

**Comment by G-Rath** [comment]
Beep boop ignore me - this was [fixed at some point](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/typescript-estree/src/ts-estree/ts-estree.ts#L1315).
